codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java

```diff
@@ -92,12 +92,15 @@ protected void encode(ChannelHandlerContext ctx, Object msg, List<Object> out) t
             buf = ctx.alloc().buffer((int) headersEncodedSizeAccumulator);
             // Encode the message.
             encodeInitialLine(buf, m);
-            encodeHeaders(m.headers(), buf);
             headersEncodedSizeAccumulator = HEADERS_WEIGHT_NEW * padSizeForAccumulation(buf.readableBytes()) +
```

Comment:
This needs to happen after the headers are encoded.

Reply:
doh!